app/src/main/java/org/fossasia/pslab/fragment/ControlFragmentAdvanced.java

```diff
@@ -83,6 +85,85 @@ public void onClick(View v) {
         etwidgetControlAdvanced10.init(getContext(), 0.1, 0.0, 1.0);
         etwidgetControlAdvanced11.init(getContext(), 1.0, 0.0, 360.0);
 
+        buttonControlAdvanced1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```

Comment:
catch ```NumberFormatException``` here And check for ```sciencelab.isConnected && "SINE".equals(wavetypeW1)``` before calling scienceLab methods. Always use ```"string".equal()``` to avoid null exception.